File: 03-Empresa/cpp/clase_trabajador.cpp
#include "clase_trabajador.h"

Trabajador::Trabajador(int numeroTrabajador, string nombre, string paterno, string materno, int horasExtra, double sueldoBase, int anioIngreso) {
    this->numeroTrabajador = numeroTrabajador;
    this->nombre = nombre;
    this->paterno = paterno;
    this->materno = materno;
    this->horasExtra = horasExtra;
    this->sueldoBase = sueldoBase;
    this->anioIngreso = anioIngreso;
};

int Trabajador::getNumeroTrabajador() {
    return numeroTrabajador;
}

void Trabajador::setNumeroTrabajador(int numeroTrabajador) {
    this->numeroTrabajador = numeroTrabajador;
}

string Trabajador::getNombre() {
    return nombre;
}

void Trabajador::setNombre(string nombre) {
    this->nombre = nombre;
}

string Trabajador::getPaterno() {
    return paterno;
}

void Trabajador::setPaterno(string paterno) {
    this->paterno = paterno;
}

string Trabajador::getMaterno() {
    return materno;
}

void Trabajador::setMaterno(string materno) {
    this->materno = materno;
}

int Trabajador::getHorasExtra() {
    return horasExtra;
}

void Trabajador::setHorasExtra(int horasExtra) {
    this->horasExtra = horasExtra;
}

double Trabajador::getSueldoBase() {
    return sueldoBase;
}

void Trabajador::setSueldoBase(double sueldoBase) {
    this->sueldoBase = sueldoBase;
}

int Trabajador::getAnioIngreso() {
    return anioIngreso;
}

void Trabajador::setAnioIngreso(int anioIngreso) {
    this->anioIngreso = anioIngreso;
}

double Trabajador::calcularSueldo() {
    double total = 0.0;
    total = sueldoBase;
    total += sueldoBase * ((2022 - anioIngreso) * PRESTACION_ANUAL);
    total += MONTO_H_EXTRA * horasExtra;
    return total;
}

void Trabajador::operator= (const Trabajador& t){
    this->numeroTrabajador = t.numeroTrabajador;
    this->nombre = t.nombre;
    this->paterno = t.paterno;
    this->materno = t.materno;
    this->horasExtra = t.horasExtra;
    this->sueldoBase = t.sueldoBase;
    this->anioIngreso = t.anioIngreso;
}

File: 03-Empresa/python/clase_trabajador.py
class Trabajador:
    MONTO_H_EXTRA = 276.5;
    PRESTACION_ANUAL = 0.03;
    numeroTrabajador = 0;
    nombre = "";
    paterno = "";
    materno = "";
    horasExtra = 0.0;
    sueldoBase = 0;
    anioIngreso = 0;

    def __init__(self, numeroTrabajador = 0, nombre = "", paterno = "", materno = "", horasExtra= 0.0, sueldoBase= 0, anioIngreso = 0):
        self.numeroTrabajador = numeroTrabajador
        self.nombre = nombre
        self.paterno = paterno
        self.materno = materno
        self.horasExtra = horasExtra
        self.sueldoBase = sueldoBase
        self.anioIngreso = anioIngreso

    def getNumeroTrabajador(self):
        return self.numeroTrabajador

    def setNumeroTrabajador(self, numeroTrabajador):
        self.numeroTrabajador = numeroTrabajador

    def getNombre(self):
        return self.nombre

    def setNombre(self, nombre):
        self.nombre = nombre

    def getPaterno(self):
        return self.paterno

    def setPaterno(self, paterno):
        self.paterno = paterno

    def getMaterno(self):
        return self.materno

    def setMaterno(self, materno):
        self.materno = materno

    def getHorasExtra(self):
        return self.horasExtra

    def setHorasExtra(self, horasExtra):
        self.horasExtra = horasExtra

    def getSueldoBase(self):
        return self.sueldoBase

    def setSueldoBase(self, sueldoBase):
        self.sueldoBase = sueldoBase

    def getAnioIngreso(self):
        return self.anioIngreso

    def setAnioIngreso(self, anioIngreso):
        self.anioIngreso = anioIngreso

    def calcularSueldo(self):
        total = 0.0;
        total = self.sueldoBase;
        total += self.sueldoBase * ((2022 - self.anioIngreso) * self.PRESTACION_ANUAL);
        total += self.MONTO_H_EXTRA * self.horasExtra;
        return total;

    def __str__(self):
        string = "#: " + str(self.numeroTrabajador) + ", Nombre: " + str(self.nombre) + " " + str(self.paterno) + " " + str(self.materno) + ", Horas extra: " + str(self.horasExtra) + ", Sueldo base: " + str(self.sueldoBase) + ", Anio de ingreso: " + str(self.anioIngreso)
        return string
